src/cli/invocation/command/support/YargSaver.ts

```diff
@@ -0,0 +1,492 @@
+        if (doesSomething(this.handleInstructions) && !this.commandDemanded) {
+            return []; // ok. does something
+        }
+        if (!doesSomething(this.handleInstructions) && this.commandDemanded && this.parameters.length === 0) {
+            return []; // ok. does nothing
+        }
+        return [{
+            complaint: `Does it want to do something or not? Handler says ${
+                doesSomething(this.handleInstructions)}; demandCommand says ${
+                !this.commandDemanded}; parameters says ${this.parameters.length > 0}`, contexts: [],
+        }];
+    }
+
+}
+
+function putInContext(additionalContext: string) {
+    return (ve: ValidationError) => ({
+        complaint: ve.complaint,
+        contexts: [additionalContext, ...ve.contexts],
+    });
+}
+
+function handleFunctionFromInstructions(instr: HandleInstructions):
+    (argObject: any) => Promise<any> | undefined {
+    if (instr === DoNothing) {
+        return async args => {
+            process.stdout.write("I was told to do nothing. Args: " + stringify(args));
+        };
+    }
+    return instr.fn;
+}
+
+function hasPositionalArguments(ys: YargSaverCommand): ys is YargSaverPositionalCommand {
+    return (ys as any).commandLine.positionalArguments.length > 0;
+}
+
+function whyNotCombine(yss: YargSaverCommand[]): ValidationError[] {
+    const commonName = yss[0].commandName;
+    const reasons: ValidationError[] = [];
+    if (yss.some(hasPositionalArguments)) {
+        reasons.push({ complaint: "Cannot combine commands with positional arguments", contexts: [commonName] });
+    }
+    const yscws = yss as Array<YargSaverCommandWord | YargSaverPositionalCommand>;
+    const completeCommands = yscws.filter(ys => !ys.demandCommand);
+    if (completeCommands.length > 1) {
+        reasons.push({
+            complaint: "There are two complete commands. Descriptions: " + completeCommands.map(c => c.description).join("; "),
+            contexts: [commonName],
+        });
+    }
+    const meaningfulInstructions = yscws.filter(ys => ys.isRunnable);
+    if (meaningfulInstructions.length > 1) {
+        reasons.push({
+            complaint: "There are two functions to respond to",
+            contexts: [commonName],
+        });
+    }
+
+    return reasons;
+}
+
+function combine(yss: YargSaverCommand[]) {
+    // assumption: whyNotCombine has been called, so several things are guaranteed
+    const yswcs = yss as YargSaverCommandWord[];
+    const one = yswcs[0];
+
+    const realCommand = yswcs.find(ys => ys.isRunnable) || {
+        handleInstructions: DoNothing,
+        parameters: [] as CommandLineParameter[],
+    };
+
+    return new YargSaverCommandWord(one.commandLine,
+        _.uniq(yswcs.map(y => y.description)).join("; or, "),
+        realCommand.handleInstructions,
+        {
+            nestedCommands: _.flatMap(yswcs.map(ys => ys.nestedCommands)),
+            parameters: realCommand.parameters,
+        },
+    );
+
+}
```

Comment:
This file is long. Is there any natural way to break it up without breaking encapsulation?